(issue)Made Game Over Play Again Button Dark Red

xutopia opened the pull request:
- conditionally inject the button's background color style based on win/lose

xutopia committed:
utilize style injection to set different background color based on win-lose condition

CameronSamuels commented:
Thanks for your contributions! I really enjoy the help.

Your code does work the way we intended, but I have a few suggestions before we merge (which I can do if you want me to).

First, I think that the dark red color for the button (`#600000`) should be the hover color (`#960000`), and then the hover color should be a little bit lighter (`#a00000`). This looks nicer than the current color scheme, plus it matches the intensity of the color to the green color scheme.

Second, the CSS file (`battle.css`) still has the coloring from before. This should be removed because it is unneeded.

Third, the styles need to be injected for endless mode. Right now, it is still a green button. We need to use the same code to inject the styles when you lose ("defeat") (so red).

Thanks for the help! You will be added to the credits when we merge.

xutopia commented:
@CameronSamuels Thanks for the suggestions, they definitely help me become a better developer.

I can take care of the points you mentioned. It's pretty straightforward, I'll commit the changes, and this PR will be updated.

CameronSamuels commented:
It’ll be my pleasure to let you do it. Like you said, it helps you be a better developer. You can go ahead. Thanks!

xutopia committed:
update endless button style to match the dark red, inject button style to defeat overlay in endless mode, color tweaks

CameronSamuels commented on b/endless/endless.css:
```diff
@@ -70,7 +70,7 @@ h2 {
 	}
 }
 .btn {
-	background: #64DD17;
```
Thanks for your contributions! I love seeing people want to contribute to our projects.

So about my comment. I think you mistakenly made the button be **red** when you **have** enough points.

I am guessing that you wanted to update the not enough button's color to the new red color scheme (that is a great idea).

So I like the idea of updating the red in the endless mode selector, even if you did this by accident. Thanks!

So the problem is that you changed the green to the red so when you have enough, it is red, and when you don't have enough, it is the old red.

If you did this purposely, I don't really agree with it because the not enough button still has the outdated red colors. Also, the play button has the red color which in my opinion, symbolizes not enough, or a bad thing like cancel or disabled.

### Solution
So if you want to fix this yourself, that would be wonderful: revert the changes of the background color for `.btn`, and then change the background colors of `#ne` to the new red (what you did to `.btn` in this commit).

If you do not want to do this yourself, I am glad doing it for you.

Thanks a lot!

xutopia committed:
"revert the background color for .btn to green, update ne color to red"

xutopia commented:
Apologies about the wrong button color changes. I didn't even realize there was a **not even points** portion.
Actually, I'm still not quite sure how to obtain the **not enough points** button. I can't seem to get to that state either locally or on the deployed site.

CameronSamuels commented:
You did this correctly.

Sorry for the confusion, so to clear things up, I'll explain:

1. The player first earns redbacks by playing (you can use the dev console to enter (`localStorage.redbacks = number`).
2. The player then chooses to play endless mode, so they select characters.
3. They notice they do not have enough redbacks (27) to play using only the Archer A (the first character in Aonarchy)
4. The button turns red and says "not enough (27)" because they do not have enough (to do this, the game hides the green button (`#sp`) and shows the red button (`#ne`) which in the css, is red)
5. They go back to normal play and earn more redbacks
6. They go back to endless mode and select the Archer A and the button turns green and says "Play (27)"

VishwamR commented:
We needed that

>

VishwamR commented:
We've needed that

cameronsamuels committed:
(improvement)Improved injectStyle input parameters

- Reordered css styling to have the `:hover` after the normal style - looks nicer and is better for the quick brain
- Prettified the css code (added spacing between {}) - looks nicer
- Removed semi-colons in the css - not need because it is 1 property
- Removed blank lines above injectStyles() - looks more consistent
- Changed `background-color` to `background` - it is shorter and uses less code, and it is better for this purpose

cameronsamuels committed:
(improvement)Updated injectStyles()

- Removed empty lines above and below - more consistent
- Changed param "rule" to "rules" - a more appropriate name
- Changed `var div ...` to `var style = document.createElement('style')` - don't need this unnecessary div, so just changed it to a style tag
- Added the id to the style element - this is needed as it was removed when creating a style element instead of a div element
- Added the innerHTML to style - this used to have a `&shy;` which was unneeded, and it had the `<style ...` which is the new element created instead of a div
- Added the element to the head instead of the body

CameronSamuels commented:
@xutopia, I have added some commits to your branch `issue-106`. The 2 commits have a detailed description/list of the changes with explanations of why I did them.

I hope these comments will help you understand more coding knowledge.

You can tell me when you want me to merge this pr. I will then add you to the credits afterwards.

Thanks for helping!!

xutopia commented:
Thank you @CameronSamuels so much for your detailed commit messages. They are very helpful for me in learning best practices.
I'm curious as to why in your `(improvement)Updated injectStyles()` commit, you made the change to add the style element to the `head` instead of `body`?

It would be an honor if you could merge my pr at your earliest convenience :)

Thanks again for this amazing opportunity to contribute!

CameronSamuels commented:
@xutopia, the reason why I added the style element to the head is because styles are meant to be in the head.

It is recommended to put the `<style>` in the head because the head is meant for elements not on the page (for example, `<title>`, `<meta>`, etc) and the body is meant for elements on the page (`<h1>`, `<p>`, etc). The style element is not shown on the screen (talking about the element itself) so it should not be in the body.

\*The `<script>` tag is an exception to these recommendations and people sometimes put it in the `<head>` or the `<body>`.

Hope these tips help!!

cameronsamuels committed:
Merge branch 'master' into issue-106

CameronSamuels commented:
I merged this pr! Great job, @xutopia! I will add you to the credits as soon as possible.
